express/parsers/apps/aiida/formats/zip.py

```diff
@@ -0,0 +1,84 @@
+import esse
+
+from express.parsers.formats.zip import BaseZipParser
+from express.parsers.formats.jsonfile import BaseJSONParser
+
+
+SUPPORTED_ARCHIVE_VERSION = '0.9'
+SUPPORTED_AIIDA_VERSION = '1.4.2'
+
+ES = esse.ESSE()
+SCHEMA_MATERIAL = ES.get_schema_by_id('material')
+
+
+class AiidaZipParser(BaseZipParser):
+
+    def __init__(self, *args, **kwargs):
+        self.json_parser = BaseJSONParser
+        super(AiidaZipParser, self).__init__(*args, **kwargs)
+
+    def structures(self):
+        structures = []
+
+        with self.zip_file as source:
+            metadata = self.json_parser.loads(source.read('metadata.json'))
+            data = self.json_parser.loads(source.read('data.json'))
+
+            # version check
+            assert metadata['aiida_version'] == SUPPORTED_AIIDA_VERSION
+            assert metadata['export_version'] == SUPPORTED_ARCHIVE_VERSION
+
+            # gather structure nodes
+            nodes = data['export_data']['Node']
+            structure_nodes = {pk: node for (pk, node) in nodes.items()
+                               if node['node_type'] == 'data.structure.StructureData.'}
+
+            for pk in structure_nodes:
+                export_data = data['export_data']['Node'][pk]
+                node_attributes = data['node_attributes'][pk]
+                structures.append(self._parse_structure_node_attributes(export_data, node_attributes))
+
+        return structures
+
+    @staticmethod
+    def _parse_structure_node_attributes(export_data, attributes):
+        """Parse an AiiDA StructureData node."""
```

Comment:
(1) We should declare parameters and output in docstring
(2) Nit: docstring format is different than elsewhere in the code